### DISCUSSION TITLE
Include Alpinejs

### DISCUSSION THREAD

**Discussion by tiger360** [Tampermonkey/tampermonkey#1507] [Ideas]
Hey,
Is it possible to include Alpinejs library to handle the DOM and inject elements on it ?

Thanks

**Comment by 7nik** [comment]
Of course. Use `@require`.
```js
// ==UserScript==
// @name         New Userscript
// @namespace    http://tampermonkey.net/
// @version      0.1
// @description  try to take over the world!
// @author       You
// @match        https://github.com/*
// @icon         https://www.google.com/s2/favicons?sz=64&domain=github.com
// @require      https://unpkg.com/alpinejs@3.x.x/dist/cdn.min.js
// @grant        none
// ==/UserScript==

// repository name
const el = document.querySelector("main #repository-container-header h2");
console.log(el);
el.insertAdjacentHTML("beforeend", `<span x-data="{ message: 'I ❤️ Alpine' }" x-text="message"></span>`);
```

**Reply by mfaizsyahmi** [reply]
The example worked, but I'm having trouble getting alpinejs' elements to teleport to places on the base website, which is being created/destroyed based on the url pathnames. basically the framework the site uses renders the main part of the site based on routes.

Any suggestions to make alpine latch onto the other framework's dom properly, and survive the parent dom being created/destroyed?

**Reply by 7nik** [reply]
I usually make wrappers around [MutationSummary](https://github.com/rafaelw/mutation-summary/blob/master/Tutorial.md): [example 1](https://github.com/7nik/userscripts/blob/master/pixiv%20img%20size%20and%20pixiv.me%20link.user.js#L218-L225), [example 2](https://github.com/evazion/translate-pixiv-tags/blob/master/translate-pixiv-tags.user.js#L2422-L2432)
